### PR TITLE
pare down insecure

### PR DESCRIPTION
**Pull request by cmr**
This weren't really being used anymore to change behavior.

### CODE REVIEW

File: src/lib/coda_base/secure_random.ml [hunk 1]
```diff
@@ -1,5 +1,3 @@
 open Core
 
-let () = assert Insecure.randomness
```
Comment (imeckler) [inline src/lib/coda_base/secure_random.ml]: This was a kind of formalized TODO to make this use dev/urandom or some such. So we should make sure to do that or have a blocking issue or something before removing this

**Comment by cmr** [comment]
@imeckler Can you write an issue somewhere about `signature_hash_function`? Will `vrf_threshold_check` be fixed by the new threshold function?

**Comment by imeckler** [comment]
Looks good -- you can remove the signature_hash_function thing. And yes regarding vrf_threshold check :)

File: src/lib/coda_base/secure_random.ml (hunk 1 above)
Comment (enolan) [reply src/lib/coda_base/secure_random.ml]: We want to use `getrandom` on Linux to wait for the system entropy pool to come up.

Comment (cmr) [reply src/lib/coda_base/secure_random.ml]: @enolan I kinda want to just use https://github.com/rust-random/getrandom. Objections?

**Pull request merged**